tensorboard/webapp/runs/store/runs_reducers.ts

```diff
@@ -180,29 +181,36 @@ const dataReducer: ActionReducer<RunsDataState, Action> = createReducer(
     };
   }),
   on(runsActions.fetchRunsSucceeded, (state, {runsForAllExperiments}) => {
-    let {nextGroupColorIndex} = state;
-    const defaultColor = new Map(state.defaultColor);
+    const groupToColor = new Map(state.groupToColor);
+    const defaultRunColor = new Map(state.defaultRunColor);
 
-    runsForAllExperiments
-      .filter((run) => !Boolean(defaultColor.get(run.id)))
-      .forEach((run) => {
-        const color = CHART_COLOR_PALLETE[nextGroupColorIndex];
-        nextGroupColorIndex =
-          (nextGroupColorIndex + 1) % CHART_COLOR_PALLETE.length;
-        defaultColor.set(run.id, color);
-      });
+    const groups = groupRuns(
+      state.groupBy,
+      runsForAllExperiments,
+      state.runIdToExpId
+    );
+    Object.entries(groups).forEach(([groupId, runs]) => {
+      const color =
+        groupToColor.get(groupId) ??
+        CHART_COLOR_PALLETE[groupToColor.size % CHART_COLOR_PALLETE.length];
+      groupToColor.set(groupId, color);
+
+      for (const run of runs) {
+        defaultRunColor.set(run.id, color);
```

Comment:
so.. after grouping (by exp/regex etc). The colors for each run would just change. Even when users go back to color by run, it would just stay there?

I think the naming "default" is confusing me.